x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/optimizer/Optimizer.java

```diff
@@ -976,14 +979,20 @@ private Expression literalToTheRight(BinaryOperator<?, ?, ?, ?> be) {
         protected Expression rule(Expression e) {
             if (e instanceof And) {
                 return propagate((And) e);
+            } else if (e instanceof Or) {
+                return propagate((Or) e);
             }
             return e;
         }
 
         // combine conjunction
         private Expression propagate(And and) {
             List<Range> ranges = new ArrayList<>();
-            List<BinaryComparison> equals = new ArrayList<>();
+            // Only equalities, not-equalities and inequalities with a foldable .right are extracted separately;
+            // the others go into the general 'exps'.
+            List<BinaryComparison> frEquals = new ArrayList<>();
```

Comment:
Right, it wasn't a great solution.. I've gone ahead and adopted Costin's solution to simply remove the prefix. I've considered the `foldableRight` prefix, but thought that having it would slightly reduce the var distinguishability due to lenght (i.e. `foldableRightEqual` vs `foldableRightNotEqual` vs `foldableRightInequalities` etc.); and `fold` isn't super clear imo: foldable vs folded vs..?
But no strong feelings about it.